metricbeat/mb/registry.go

```diff
@@ -300,11 +300,33 @@ func (r *Register) Modules() []string {
 	r.lock.RLock()
 	defer r.lock.RUnlock()
 
+	var dups = map[string]bool{}
+
+	// For the sake of compatibility, grab modules the old way as well, right from the modules map
 	modules := make([]string, 0, len(r.modules))
 	for module := range r.modules {
+		dups[module] = true
 		modules = append(modules, module)
 	}
 
+	// List also modules from secondary sources
+	if source := r.secondarySource; source != nil {
+		sourceModules, err := source.Modules()
+		if err != nil {
+			logp.L().Errorf("failed to get modules from secondary source: %s", err)
+		}
+		modules = append(modules, sourceModules...)
```

Comment:
We should add them to `dups` too to avoid duplications later. There can be "mixed" modules with light and native modules (as `aws.elb` that is going to be a light module based on `aws.cloudwatch`).